cmd/mint/login.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+
+	"github.com/pkg/errors"
+	"github.com/skratchdot/open-golang/open"
+
+	"github.com/rwx-research/mint-cli/internal/cli"
+
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+)
+
+var (
+	DeviceName string
+
+	loginCmd = &cobra.Command{
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// try to collect the device name if one is not provided
+			if DeviceName == "" {
+				prompt := promptui.Prompt{
+					Label:   "Device Name",
+					Default: defaultDeviceName(),
+					Validate: func(s string) error {
+						if s == "" {
+							return errors.New("Device name must be provided")
+						}
+
+						return nil
+					},
+				}
+				deviceName, err := prompt.Run()
+				if err != nil {
+					return err
+				}
+
+				DeviceName = deviceName
+			}
+
+			err := service.Login(
+				cli.LoginConfig{
+					DeviceName:         DeviceName,
+					AccessTokenBackend: accessTokenBackend,
+					Stdout:             os.Stdout,
+					OpenUrl:            open.Run,
+				},
+			)
+			if err != nil {
+				return err
+			}
+
+			return nil
+
+		},
+		Short: "Authorize subsequent commands on this device with RWX Cloud",
+		Use:   "login [flags]",
+	}
+)
+
+func init() {
+	// A different host can only be set over the environment
+	mintHost = os.Getenv("MINT_HOST")
+	if mintHost == "" {
+		mintHost = "cloud.rwx.com"
+	}
```

Comment:
I think Pierre just made a change which sets this up in root.cmd, so this might be unnecessary now

Reply:
Yea, I had copied over some of his WIP to my branch originally but in the very complicated rebase, I missed this. Thanks!